client/index.html

```diff
@@ -58,6 +58,7 @@ <h1 class="title">The Lounge is loading…</h1>
 				<div id="chat-container" class="window">
 					<div id="chat"></div>
 					<form id="form" method="post" action="">
+						<div id="connection-error"><span>Connection failed: <span id="connection-error-reason">??</span>. Refresh the page to reconnect.</span></div>
```

Comment:
It would be great if this could be displayed above the text input instead of on it. One of the most annoying consequence to the bug you're fixing is that connection is often lost while typing a message (while your PR is 👏 👏 when this happens while reading messages).
Until the full reconnection is fixed, having this outside the form input makes it possible to copy into clipboard what I was typing, refresh, paste. Not ideal, but as you say ideally this whole thing would be fixed someday anyway, and in the meanwhile, it'll help.

What do you think?

You might have to disable the sending of the message though, but that is also probably necessary in the current solution as one can still press <kbd>Enter</kbd> (but since it has no effect...)